packages/zilliqa-js-util/test/unit.spec.ts

```diff
@@ -0,0 +1,46 @@
+import BN from 'bn.js';
+import * as denom from '../src/unit';
+
+describe('Unit utils', () => {
+  it('should convert Qa to Zil', () => {
+    const qa = new BN(1_000_000_000_000);
+    const expected = '1';
+
+    expect(denom.fromQa(qa, denom.Units.Zil)).toEqual(expected);
+  });
+
+  it('should convert Qa to Li', () => {
+    const qa = new BN(1_000_000);
+    const expected = '1';
+
+    expect(denom.fromQa(qa, denom.Units.Li)).toEqual(expected);
+  });
+
+  it('should convert Li to Qa', () => {
+    const li = new BN(1);
+    const expected = new BN(1_000_000);
+
+    expect(denom.toQa(li, denom.Units.Li).eq(expected)).toBeTruthy();
+  });
+
+  it('should convert Zil to Qa', () => {
+    const zil = new BN(1);
+    const expected = new BN(1_000_000_000_000);
+
+    expect(denom.toQa(zil, denom.Units.Zil).eq(expected)).toBeTruthy();
+  });
+
+  it('fromQa should should work for negative numbers', () => {
```

Comment:
Are there any reasons why we are supporting negative numbers? This is correct but i'm wondering if it is needed.

Reply:
I think the use cases are pretty narrow, but since this is an adaptation of the implementation in https://github.com/ethjs/ethjs-unit, the cost is virtually nil to support negative numbers.